docs/performance/selects.rst

```diff
@@ -110,6 +110,63 @@ to the original result.
    based sampling.
 
 
+.. _downsampling-timestamp-binning:
+
+Downsampling with ``DATE_BIN``
+==============================
+
+For improved downsampling using time-bucketing and resampling, the article
+`resampling time-series data with DATE_BIN`_ shares patterns how to
+group records into time buckets and resample the values.
+
+This technique will improve query performance by reducing the amount of data
+needed to be transferred, by decreasing its granularity on the time dimension.
+Most often, this is applied when querying live system metrics data using
+visualization or dashboarding tools like Grafana and friends.
+
+.. code-block:: sql
+
+    SELECT ts_bin,
+           battery_level,
+           battery_status,
+           battery_temperature
+    FROM (
+      SELECT DATE_BIN('5 minutes'::INTERVAL, "time", 0) AS ts_bin,
+             battery_level,
+             battery_status,
+             battery_temperature,
+             ROW_NUMBER() OVER (PARTITION BY DATE_BIN('5 minutes'::INTERVAL, "time", 0) ORDER BY "time" DESC) AS "row_number"
+      FROM doc.sensor_readings
+    ) x
+    WHERE "row_number" = 1
+    ORDER BY 1 ASC
+
+
+.. _downsampling-lttb:
+
+Downsampling with LTTB
+======================
+
+`Largest Triangle Three Buckets`_ is a downsampling method that tries to retain
+visual similarity between the downsampled data and the original dataset using
+considerably fewer data points.
+
+The article about `advanced downsampling with the LTTB algorithm`_ explains how
+to use LTTB with CrateDB. This technique is mostly used for the same purposes
+like other downsampling procedures, where, in this case, retaining essential
+details is important for proper visual graph analysis.
+
+.. code-block:: sql
+
+    WITH downsampleddata AS
+    (	SELECT lttb_with_parallel_arrays(
+        array(SELECT n FROM demo ORDER BY n),
+        array(SELECT reading FROM demo ORDER BY n)
+        ,100) AS lttb)
+    SELECT unnest(lttb['0']) as n, unnest(lttb['1']) AS reading
+    FROM downsampleddata;
```

Comment:
This probably also has a formatting hiccup, @hammerhead?

Reply:
Reformatted using https://sqlformat.org/, then edited manually a bit. Is it a good choice?
```suggestion
    WITH downsampleddata AS
      (SELECT lttb_with_parallel_arrays(
      array(SELECT n FROM demo ORDER BY n),
      array(SELECT reading FROM demo ORDER BY n), 100) AS lttb)
    SELECT unnest(lttb['0']) AS n,
           unnest(lttb['1']) AS reading
    FROM downsampleddata;
```

Reply:
I would love to have a standard formatting tool that we can agree on. Like (at least in our team) we settled on [black](https://pypi.org/project/black/) for Python, and it's considered the truth™️ that we all stick to. Without the "then edited manually a bit" part you mentioned, as that kind of defeats the purpose.

For community posts, I don't dare to be the one nit-picking on formatting all the time, but as soon as we are talking about the main documentation, there needs to be a consistent style and quality of formatting. With formatting being such a subjective topic, only a tool we all abide to can do that job.

Reply:
I agree we need to level up. Future improvements, also expanding into cratedb-examples, will/should/may probably include:
- Using `ruff` across the board.
- Use `black[jupyter]` for canonicalizing all the Jupyter Notebooks we introduced recently.
- For prose, we are already using [Vale](https://github.com/errata-ai/vale) and [proselint](https://pypi.org/project/proselint/).
- More potential linters: https://markuplint.dev/, https://textlint.github.io/
- For other file formats, https://github.com/super-linter/super-linter has been probed at https://github.com/crate/infrastructure/pull/2688.
- For SQL, the formatter at https://sqlformat.org/ is actually from [sqlparse](https://pypi.org/project/sqlparse/), which we are already using at other places.

_=> We probably want to break this out into a different discussion._

Reply:
I've diverted this into another issue.

- https://github.com/crate/tech-writing/issues/396